mojo/stdlib/src/builtin/float_literal.mojo

```diff
@@ -61,7 +61,7 @@ struct FloatLiteral[value: __mlir_type.`!pop.float_literal`](
         Args:
             value: The IntLiteral value.
         """
-        out = __type_of(result)()
+        result = __type_of(result)()
```

Comment:
Do we have pre-existing tests for this? It's not clear how this worked before. If not, can you add some tests, please?